File: backend/src/main/java/com/example/E_commerce/Platform/services/facade/CartItemService.java
package com.example.E_commerce.Platform.services.facade;

import com.example.E_commerce.Platform.entities.Cart;
import com.example.E_commerce.Platform.entities.CartItem;

import java.util.List;

public interface CartItemService {
    CartItem save(CartItem cartItem);

    void deleteById(Long id);

    CartItem findById(Long id);

    List<CartItem> findAll();

    CartItem update(Long id, CartItem cartItem);
    public CartItem addCartItemToCart(Long userId, Long productId, int quantity);
    CartItem updateCartItemQuantity(Long id, int quantity);

}

File: frontend/src/admin/services/cartItemService.ts
import {CartItem} from "@/types";
import api from "@/axiosConfig.ts";

const API_URL = "/cart-item";

export const cartItemService = {

    async addCartItemToCart(
        userId: number,
        productId: number,
        quantity: number
    ): Promise<CartItem> {

        return  await api.post(
            `${API_URL}/userId/${userId}/productId/${productId}`,
            null,
            {
                params: {
                    quantity
                }
            }
        );


    },
    async updateCartItemQuantity(id:number, quantity:number):Promise<any>{
        const response = await api.put(`${API_URL}/update/cartItemId/${id}`,null,
            {
                params: {
                    quantity
                }
            });
        return response.data;
    }


}